test/testframe.c

```diff
@@ -11,368 +11,504 @@
- *
- * Return: Void
- *    exit EXIT_FAILURE if error is encountered.
+ * Parse command line information
  */
-void
+herr_t
 TestParseCmdLine(int argc, char *argv[])
 {
-    bool skipped_all = false;
-    int  ret_code;
+    herr_t ret_value = SUCCEED;
 
     while ((void)argv++, --argc > 0) {
         if ((strcmp(*argv, "-verbose") == 0) || (strcmp(*argv, "-v") == 0)) {
             if (argc > 0) {
                 --argc;
                 ++argv;
-                ParseTestVerbosity(*argv);
+
+                if (ParseTestVerbosity(*argv) < 0) {
+                    ret_value = FAIL;
+                    goto done;
+                }
             }
             else {
-                TestUsage();
-                exit(EXIT_FAILURE);
+                ret_value = FAIL;
+                goto done;
             }
         }
         else if (((strcmp(*argv, "-exclude") == 0) || (strcmp(*argv, "-x") == 0))) {
             if (argc > 0) {
                 --argc;
                 ++argv;
-                SetTest(*argv, SKIPTEST);
+
+                if (SetTest(*argv, SKIPTEST) < 0) {
+                    ret_value = FAIL;
+                    goto done;
+                }
             }
             else {
-                TestUsage();
-                exit(EXIT_FAILURE);
+                ret_value = FAIL;
+                goto done;
             }
         }
         else if (((strcmp(*argv, "-begin") == 0) || (strcmp(*argv, "-b") == 0))) {
             if (argc > 0) {
                 --argc;
                 ++argv;
-                SetTest(*argv, BEGINTEST);
+
+                if (SetTest(*argv, BEGINTEST) < 0) {
+                    ret_value = FAIL;
+                    goto done;
+                }
             }
             else {
-                TestUsage();
-                exit(EXIT_FAILURE);
+                ret_value = FAIL;
+                goto done;
             }
         }
         else if (((strcmp(*argv, "-only") == 0) || (strcmp(*argv, "-o") == 0))) {
             if (argc > 0) {
-                unsigned Loop;
-
                 --argc;
                 ++argv;
 
-                /* Skip all tests, then activate only one. */
-                if (!skipped_all) {
-                    for (Loop = 0; Loop < Index; Loop++)
-                        Test[Loop].SkipFlag = 1;
-                    skipped_all = true;
-                } /* end if */
-                SetTest(*argv, ONLYTEST);
+                if (SetTest(*argv, ONLYTEST) < 0) {
+                    ret_value = FAIL;
+                    goto done;
+                }
             }
             else {
-                TestUsage();
-                exit(EXIT_FAILURE);
+                ret_value = FAIL;
+                goto done;
             }
         }
         else if ((strcmp(*argv, "-summary") == 0) || (strcmp(*argv, "-s") == 0))
-            Summary = 1;
-        else if (strcmp(*argv, "-enable-error-stack") == 0)
```

Comment:
To prevent hiding of problems, enabled error stacks by default for testframe.c tests. This includes testhdf5, so some new error stack suppressions had to be added to some tests.